File: Shopee project/Shopee2/src/main/java/fptu/shopee2/Model/ProductPakage/Category.java
package fptu.shopee2.Model.ProductPakage;

import jakarta.persistence.*;

import java.util.Set;

@Entity
@Table(name = "category")
public class Category {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_category")
    private int idCategory;

    @Column(name = "name_category", nullable = false)
    private String nameCategory;

    // Mối quan hệ Many-to-One với chính bảng Category để thiết lập danh mục cha
    @ManyToOne
    @JoinColumn(name = "parent_id") // parent_id là khóa ngoại tham chiếu đến idCategory của bảng cha
    private Category category;

    // Mối quan hệ One-to-Many để lấy tất cả danh mục con của danh mục hiện tại
    @OneToMany(mappedBy = "category", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private Set<Category> categories;

    @OneToMany(mappedBy = "category", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private Set<CategoryAttribute> categoryAttributes;

    @OneToMany(mappedBy = "category", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private Set<Product> products;

    public int getIdCategory() {
        return idCategory;
    }

    public void setIdCategory(int idCategory) {
        this.idCategory = idCategory;
    }

    public String getNameCategory() {
        return nameCategory;
    }

    public void setNameCategory(String nameCategory) {
        this.nameCategory = nameCategory;
    }

}

File: Shopee Project By .Net/Code Shopee/Model_Shopee_Project/Models/Category.cs
using System;
using System.Collections.Generic;

namespace Model_Shopee_Project.Models;

public partial class Category
{
    public int IdCategory { get; set; }

    public string NameCategory { get; set; } = null!;

    public int? IdParent { get; set; }

    public virtual ICollection<CategoryAttribute> CategoryAttributes { get; set; } = new List<CategoryAttribute>();

    public virtual Category? IdParentNavigation { get; set; }

    public virtual ICollection<Category> InverseIdParentNavigation { get; set; } = new List<Category>();

    public virtual ICollection<Product> Products { get; set; } = new List<Product>();
}
